llvm/lib/CAS/OnDiskHashMappedTrie.cpp

```diff
@@ -493,8 +493,8 @@ class HashMappedTrieHandle {
   LazyMappedFileRegion &getRegion() const { return *LMFR; }
 
   size_t getFlags() const { return H->Flags; }
-  uint64_t getNumSubtrieBits() const { return H->NumSubtrieBits; }
-  uint64_t getNumHashBits() const { return H->NumHashBits; }
+  uint32_t getNumSubtrieBits() const { return H->NumSubtrieBits; }
```

Comment:
Hmm, what is the actual range for these values? We seem really inconsistent about them
* `ThreadSafeHashMappedTrieBase` stores `unsigned short NumSubtrieBits`
* `ThreadSafeHashMappedTrieBase`, `ThreadSafeHashMappedTrie` and `OnDiskHashMappedTrie` constructors have `Optional<size_t> NumSubtrieBits`
* `HashMappedTrieHandle::Header` has `uint8_t NumSubtrieBits`
* `HashMappedTrieHandle::Limits` has `MaxNumSubtrieBits = 10`

If the maximum value really is 10, maybe we should use `uint8_t` everywhere?

Similarly for hash bits
* `HashMappedTrieHandle::Header` has `uint16_t NumHashBits`
* `HashMappedTrieHandle::Limits` has `MaxNumHashBits = UINT16_MAX`
* Otherwise seems to be `size_t`.

Reply:
I think the current implement just assert when you give a number that is too big to fix into HashMapTrie but the interface is very general.

The change is basically fixing we feed the return value into a uint32_t which causes clamming error on MSVC. But I guess I should just audit the size to be narrower since the generic API doesn't provide any value.